stl/inc/barrier

```diff
@@ -81,7 +81,7 @@ public:
     constexpr explicit barrier(
         const ptrdiff_t _Expected, _Completion_function _Fn = _Completion_function()) noexcept /* strengthened */
         : _Val(_One_then_variadic_args_t{}, _STD move(_Fn), _Expected << _Barrier_value_shift) {
-        _STL_VERIFY(_Val._Myval2._Current.load(memory_order_relaxed) >= 0,
+        _STL_VERIFY(_Expected >= 0 && _Expected <= max(),

```

Comment:
We need to parenthesize `max()` to defend against `<Windows.h>` macroization. (It defines `max` as a function-like macro, and the preprocessor is barely intelligent enough to avoid expansion when it sees `(max)()` instead.) Occurs below. The failure is:

```
D:\GitHub\STL\out\x64>type meow.cpp
```
```cpp
#include <Windows.h>
#include <barrier>
int main() {}
```
```
D:\GitHub\STL\out\x64>cl /EHsc /nologo /W4 /std:c++latest /c meow.cpp
meow.cpp
D:\GitHub\STL\out\x64\out\inc\barrier(85): warning C4003: not enough arguments for function-like macro invocation 'max'
D:\GitHub\STL\out\x64\out\inc\barrier(97): warning C4003: not enough arguments for function-like macro invocation 'max'
D:\GitHub\STL\out\x64\out\inc\barrier(84): error C2760: syntax error: ')' was unexpected here; expected 'expression'
D:\GitHub\STL\out\x64\out\inc\barrier(187): note: see reference to class template instantiation 'std::barrier<_Completion_function>' being compiled
[...]
```

We have test coverage for this but it is currently MSVC-internal which is why the GitHub tests passed.